unified-runtime/source/adapters/level_zero/helpers/image_helpers.hpp

```diff
@@ -24,24 +28,34 @@ struct ur_bindless_mem_handle_t {
 
   ur_bindless_mem_handle_t(ze_image_handle_t zeImage,
                            const ZeStruct<ze_image_desc_t> &zeImageDesc)
-      : zeImage(zeImage), zeImageDesc(zeImageDesc){};
+      : zeImage(zeImage), zeImageDesc(zeImageDesc) {};
 
+#ifdef UR_ADAPTER_LEVEL_ZERO_V2
+  ze_image_handle_t getZeImage() const { return zeImage.get(); }
+#else
   ze_image_handle_t getZeImage() const { return zeImage; }
+#endif
+
   ze_image_desc_t &getZeImageDesc() { return zeImageDesc; }
 
 private:
-  ::ze_image_handle_t zeImage;
+
+#ifdef UR_ADAPTER_LEVEL_ZERO_V2
```

Comment:
I think you can just use v2::raii::ze_image_handle_t event in v1 adapter - i.e. just remove the ifdefs.